nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -265,13 +329,15 @@ def collate_fn(self, batch):
         contexts = [item['context_ids'] for item in batch]
         context_lengths = torch.LongTensor([item['context_length'] for item in batch])
         loss_mask = [self._build_loss_mask(item)[1:] for item in batch]
+        metadata = [item['metadata'] for item in batch]
 
-        max_length = max([len(x) for x in input_ids]) + self.tokens_to_generate
+        max_length = max(max([len(x) for x in input_ids]), max([len(x) for x in contexts]) + self.tokens_to_generate)
```

Comment:
when does context could be longer input_ids?

Reply:
It is `contexts` + `tokens_to_generate` compared to `input_ids` (`contexts` + `answers`).